docs/source/en/using-diffusers/write_own_pipeline.mdx

```diff
@@ -0,0 +1,266 @@
+
+```py
+>>> from diffusers import DDPMPipeline
+
+>>> ddpm = DDPMPipeline.from_pretrained("google/ddpm-cat-256").to("cuda")
+>>> image = ddpm(num_inference_steps=25).images[0]
+>>> image
+```
+
+That was super easy, but how did the pipeline do that? Let's breakdown the pipeline and take a look at what's happening under the hood.
+
+In the example above, the pipeline contains a UNet model and a DDPM scheduler. The pipeline denoises an image by taking random noise the size of the desired output and passing it through the model several times. At each timestep, the model predicts the *noise residual* and the scheduler uses it to predict a less noisy image. The pipeline repeats this process until it reaches the end of the specified number of inference steps. 
+
+To recreate the pipeline with the model and scheduler separately, let's write our own denoising process.
+
+1. Load the model and scheduler:
+
+    ```py
+    >>> from diffusers import DDPMScheduler, UNet2DModel
+
+    >>> scheduler = DDPMScheduler.from_pretrained("google/ddpm-cat-256")
+    >>> model = UNet2DModel.from_pretrained("google/ddpm-cat-256").to("cuda")
+    ```
+
+2. Set the number of timesteps to run the denoising process for:
+
+    ```py
+    >>> scheduler.set_timesteps(50)
+    ```
+
+3. Create some random noise with the same shape as the desired output:
+
+    ```py
+    >>> import torch
+
+    >>> sample_size = model.config.sample_size
+    >>> noise = torch.randn((1, 3, sample_size, sample_size)).to("cuda")
+    ```
+
+4. Now write a loop to iterate over the number of timesteps. At each timestep, the model does a [~`UNet2DModel.forward`] pass and returns the noisy residual. The scheduler's [`~DDPMScheduler.step`] method takes the noisy residual, timestep, and input and it predicts the image at the previous timestep. This output becomes the next input to the model in the denoising loop, and it'll repeat until it reaches the end of the `set_timesteps`.
+
+    ```py
+    >>> input = noise
+
+    >>> for t in scheduler.timesteps:
+    ...     with torch.no_grad():
+    ...         noisy_residual = model(input, t).sample
+    >>> previous_noisy_sample = scheduler.step(noisy_residual, t, input).prev_sample
+    >>> input = previous_noisy_sample
+    ```
+
+    This is the entire denoising process, and you can use this same pattern to write any diffusion system.
+
+5. The last step is to convert the denoised output into an image:
+
+    ```py
+    >>> from PIL import Image
+    >>> import numpy as np
+
+    >>> image = (input / 2 + 0.5).clamp(0, 1)
+    >>> image = image.cpu().permute(0, 2, 3, 1).numpy()[0]
+    >>> image = Image.fromarray((image * 255)).round().astype("uint8")
+    >>> image
+    ```
+
+In the next section, you'll put your skills to the test and breakdown the more complex Stable Diffusion pipeline, but it's still built from the same components and follows the same denoising pattern.
```

Comment:
Nice! I'd maybe only add 3, 4 sentences to the relationship between `set_timesteps` and `timesteps` to make it crystal clear